src/main/java/appeng/util/ItemSorters.java

```diff
@@ -54,38 +52,30 @@ public int compare( final IAEItemStack o1, final IAEItemStack o2 )
 		{
 			final AEItemStack op1 = (AEItemStack) o1;
 			final AEItemStack op2 = (AEItemStack) o2;
+			int cmp = op1.getModID().compareToIgnoreCase( op2.getModID() );
 
-			if( getDirection() == SortDir.ASCENDING )
+			if( cmp == 0 )
 			{
-				return this.secondarySort( op2.getModID().compareToIgnoreCase( op1.getModID() ), o1, o2 );
+				cmp = Platform.getItemDisplayName( o1 ).compareToIgnoreCase( Platform.getItemDisplayName( o2 ));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') ')' is not preceded with whitespace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck)